src/dataflow/stream.rs

```diff
@@ -232,7 +232,8 @@ impl<IN: ArconType> Stream<IN> {
         let mut state_dir = self.ctx.app.arcon_conf().state_dir();
         let state_id = builder.state_id();
         state_dir.push(state_id.clone());
-        let backend = builder.create_backend(state_dir);
+        let backend =
```

Comment:
Right, I see the issue here.  A mistake on my part. We will need to give the Backend a String instead of 'static str when creating it. Storing the name as String inside the backends.  Not much else has to change, we will still access the String as &str.

Reply:
But that will result into multiple clones like before

Reply:
May result in some clones in the dataflow construction. But we should still be able to access that String as ``&str`` in Backend without doing clones?

Reply:
Yeah that can be done